### ISSUE TITLE
Data & CRC errors from some archives (even if no changes were made)

### ISSUE THREAD

**Issue by carpiediem** [cthackers/adm-zip#262]
I'm having trouble finding a root cause here, so I've included the script below that shows the difference I'm seeing. Some of the archives that I run through new AdmZip() & writeZip() have no problems, while others have CRC errors in a 7zip test (and trigger errors when used in a web app). The source archives don't have any issues when I run them through 7zip's tester and I can't seem to find any pattern in what works and what doesn't...

I was originally looking for issues in `writeZip()`, until I found that `getCompressedData()` was failing for entries in a ZipFile made from the original archive.

```
const AdmZip = require('adm-zip');
const fs = require('fs');
const copyCallback = err => {
  if (err) {
    console.error('Failed to copy SDK assets to /lib. Try running `npm link emotics-attention-sdk`');
    throw err;
  }
};

function compareOriginalCopyAndRezip(path) {
  const zip = new AdmZip(path);
  const copiedPath = path.replace('.zip', '.copied.zip');
  const rezippedPath = path.replace('.zip', '.rezipped.zip');

  fs.copyFile(path, copiedPath, copyCallback);
  zip.writeZip(rezippedPath);

  const origStats = fs.statSync(path);
  const copiedStats = fs.statSync(copiedPath);
  const rezippedStats = fs.statSync(rezippedPath);

  if (copiedStats.size === origStats.size && rezippedStats.size === origStats.size) {
    console.log('All versions of of %s are the same size.', path);
  } else {
    console.log('File size has changed.', {
      original: origStats.size,
      copied: copiedStats.size,
      rezipped: rezippedStats.size
    });

    const origEntries = zip.getEntries();

    console.log('Testing entries in ZipFile made from the original file:');
    origEntries.forEach(entry => {
      if (entry.isDirectory) { return; }
      try {
        entry.getCompressedData();
      } catch (error) {
        console.log('getCompressedData() failed for', entry.name);
      }
    });
  }
}

compareOriginalCopyAndRezip('./scos/Risk_Management_-_An_Introduction_KHTUT_46.00.03.40.zip');
compareOriginalCopyAndRezip('./scos/Commodities_-_Emissions_KHTUT_42.00.00.zip');
```

Console output:

```
All versions of of ./scos/Risk_Management_-_An_Introduction_KHTUT_46.00.03.40.zip are the same size.
File size has changed. { original: 19996110, copied: 19996110, rezipped: 20000607 }
compressedData of first entry: { original: '19996110 bytes buffer',
  rezipped: '20000607 bytes buffer' }
Testing entries in ZipFile made from the original file:
getCompressedData() failed for close.htm
getCompressedData() failed for index_vid_full.css
getCompressedData() failed for index_vid_full_ie8.css
getCompressedData() failed for loader.css
... [this matches the output of 7zip's test]
```

Any ideas as to what's going on?

**Comment by carpiediem** [comment]
7zip confirms that `Commodities_-_Emissions_KHTUT_42.00.00.rezipped.zip has several data & CRC errors that none of the other ZIP files have (including the original).  I don't see any clear pattern among the files that do have errors vs. those that don't.

**Comment by sirhc9** [comment]
I'm experiencing the same issue with 7zip reporting bad CRC. Interestingly in my example I am attempting to zip 3 text files in a folder. 2 of these files have been downloaded using curl and modified with sed, these both have CRC issues. Meanwhile the third file is only generated locally and the CRC reports okay.

My code to zip these files is below.

var zip = new AdmZip("");
zip.addLocalFolder("/tmp/someFolder");
zip.writeZip("/tmp/someFolder.zip");

**Comment by 5saviahv** [comment]
It seems to be fixed with 0.4.14
